src/platform/Linux/ThreadStackManagerImpl.cpp

```diff
@@ -15,99 +15,93 @@
-
-#define LogClientError(error)                                                                                                      \
-    do                                                                                                                             \
-    {                                                                                                                              \
-        if (error != ClientError::ERROR_NONE)                                                                                      \
-        {                                                                                                                          \
-            ChipLogError(DeviceLayer, __FILE__ " %d: Otbr ClientError %d", __LINE__, static_cast<int>(error));                     \
-        }                                                                                                                          \
-    } while (0)
-
-constexpr int kDBusConnectionPollingTimeoutMS = 10;
+#include <platform/PlatformManager.h>
+#include <platform/ThreadStackManager.h>
+#include <support/CodeUtils.h>
+#include <support/SmartPointerUtility.h>
+#include <support/logging/CHIPLogging.h>
 
 namespace chip {
 namespace DeviceLayer {
 
 ThreadStackManagerImpl ThreadStackManagerImpl::sInstance;
 
-ThreadStackManagerImpl::ThreadStackManagerImpl() : mThreadApi(nullptr), mConnection(nullptr), mAttached(false) {}
+constexpr char ThreadStackManagerImpl::kDBusOpenThreadService[];
+constexpr char ThreadStackManagerImpl::kDBusOpenThreadObjectPath[];
 
-CHIP_ERROR ThreadStackManagerImpl::_InitThreadStack()
-{
-    ClientError error;
-    DeviceRole role;
-    DBusError dbusError;
-    DBusConnection * dispatchConnection;
+constexpr char ThreadStackManagerImpl::kOpenthreadDeviceRoleDisabled[];
+constexpr char ThreadStackManagerImpl::kOpenthreadDeviceRoleDetached[];
+constexpr char ThreadStackManagerImpl::kOpenthreadDeviceRoleChild[];
+constexpr char ThreadStackManagerImpl::kOpenthreadDeviceRoleRouter[];
+constexpr char ThreadStackManagerImpl::kOpenthreadDeviceRoleLeader[];
 
-    dbus_error_init(&dbusError);
-    mConnection = UniqueDBusConnection(dbus_bus_get_private(DBUS_BUS_SYSTEM, &dbusError));
+constexpr char ThreadStackManagerImpl::kPropertyDeviceRole[];
 
-    if (mConnection == nullptr)
-    {
-        dbus_error_free(&dbusError);
-        error = ClientError::ERROR_DBUS;
-        LogClientError(error);
+ThreadStackManagerImpl::ThreadStackManagerImpl() : mProxy(nullptr, g_object_unref), mAttached(false) {}
 
-        return OTBR_TO_CHIP_ERROR(error);
+CHIP_ERROR ThreadStackManagerImpl::_InitThreadStack()
+{
+    GError * gdbusError = nullptr;
+    mProxy.reset(openthread_io_openthread_border_router_proxy_new_for_bus_sync(
+        G_BUS_TYPE_SYSTEM, G_DBUS_PROXY_FLAGS_NONE, kDBusOpenThreadService, kDBusOpenThreadObjectPath, nullptr, &gdbusError));
+    auto err = wrap_unique(gdbusError, &g_error_free);
```

Comment:
If we want to avoid this slightly awkward adoption of an out parameter we could use an approach like this:

https://source.chromium.org/chromium/chromium/src/+/main:base/scoped_generic.h;l=187-188;drc=ea5a3cf75400998f40936c556ec7f28645d616cc